### PR TITLE
FMWK-55 Supporting Map and List filtering (integer only), refactoring Qualifier, AerospikeCriteria, AerospikeQueryCreator

### PR DESCRIPTION
**Pull request by agrgr**
- refactoring Qualifier switch-case
- Qualifier telescoping constructors into a Builder
- AerospikeCriteria constructors into a Builder
- Qualifier FilterOperation enum out as a separate class
- removing Criteria class
- adding support for querying by  map elements similar to findByPojoField
- increasing readability, reducing excessive code etc.
- test coverage

### CODE REVIEW

**Review by reugn — APPROVED** [review]

File: src/test/java/org/springframework/data/aerospike/sample/PersonRepository.java [hunk 1]
```diff
@@ -130,7 +132,55 @@
 
 	Page<P> findByAddressIn(List<Address> address, Pageable page);
 
-	List<P> findByMapKeysContaining(String key);
+	/**
```
Comment (roimenashe) [inline src/test/java/org/springframework/data/aerospike/sample/PersonRepository.java:135]: Please run formatting - it will align all javadoc comments.

Comment (roimenashe) [reply src/test/java/org/springframework/data/aerospike/sample/PersonRepository.java]: Pushed.

File: src/main/java/org/springframework/data/aerospike/repository/query/Criteria.java [hunk 2]
```diff
@@ -67,7 +67,11 @@ public Qualifier getCriteriaObject() {
 			for (Criteria c : this.criteriaChain) {
 				qualifiers.add(c.getCriteriaObject());
 			}
-			return new Qualifier(op, qualifiers.toArray(new Qualifier[0]));
+//			return new Qualifier(op, qualifiers.toArray(new Qualifier[0]));
```
Comment (roimenashe) [inline src/main/java/org/springframework/data/aerospike/repository/query/Criteria.java:70]: Remove comment?

**Review by roimenashe — APPROVED** [review]

**Review by ijusti — APPROVED** [review]
Is there estimates or milestones for release?

File: src/main/java/org/springframework/data/aerospike/repository/query/Criteria.java (hunk 2 above)
Comment (agrgr) [reply src/main/java/org/springframework/data/aerospike/repository/query/Criteria.java]: Criteria class is going to be removed (as discussed)

**Comment by agrgr** [comment]
> Is there estimates or milestones for release?

@ijusti, there is secondary index related work left, as far as I know

**Review by roimenashe — APPROVED** [review]

File: src/main/java/org/springframework/data/aerospike/query/FilterOperation.java [hunk 3]
```diff
@@ -0,0 +1,743 @@
...
+            Qualifier[] qs = (Qualifier[]) map.get(QUALIFIERS);
+            Exp[] childrenExp = new Exp[qs.length];
+            for (int i = 0; i < qs.length; i++) {
+                childrenExp[i] = qs[i].toFilterExp();
+            }
+            return Exp.and(childrenExp);
+        }
+
+        @Override
+        public Filter secIndexFilter(Map<String, Object> map) {
```
Comment (roimenashe) [inline src/main/java/org/springframework/data/aerospike/query/FilterOperation.java:40]: Please change `secIndexFilter` to `sIndexFilter` (it's more common in most of Aerospike documentation regarding secondary indexes).

File: src/main/java/org/springframework/data/aerospike/query/FilterOperation.java [hunk 4]
```diff
@@ -0,0 +1,743 @@
...
+    }
+
+    protected Filter collectionRange(IndexCollectionType collectionType, Map<String, Object> map) {
+        return Filter.range(getField(map), collectionType, getValue1(map).toLong(), getValue2(map).toLong());
+    }
+
+    protected Filter geoWithinRadius(IndexCollectionType collectionType, Map<String, Object> map) {
+        return Filter.geoContains(getField(map), getValue1(map).toString());
+    }
+
```
Comment (roimenashe) [inline src/main/java/org/springframework/data/aerospike/query/FilterOperation.java:740]: Remove 3 unnecessary spaces.

File: src/main/java/org/springframework/data/aerospike/query/FilterOperation.java (hunk 3 above)
Comment (agrgr) [reply src/main/java/org/springframework/data/aerospike/query/FilterOperation.java]: Done

File: src/main/java/org/springframework/data/aerospike/query/FilterOperation.java (hunk 4 above)
Comment (agrgr) [reply src/main/java/org/springframework/data/aerospike/query/FilterOperation.java]: Done

**Pull request merged**